### PR TITLE
feat: [JSX] Support short boolean syntax

### PR DESCRIPTION
**Pull request by retyui**
via: https://github.com/facebookincubator/fbt/issues/78

```js
<fbt desc="" doNotExtract>Test</fbt>

<fbt desc="">
  <fbt:param name="name" number>{person.getName()}</fbt:param>
</fbt>
```

### CODE REVIEW

**Comment by retyui** [comment]
Seems CI failed with an strange error

```log
[09:50:15] Error: ENOENT: no such file or directory, stat '/home/travis/build/facebookincubator/fbt/runtime/nonfb/DisplayGenderConst.js'
```

**Comment by jrwats** [comment]
> Seems CI failed with an strange error
> [09:50:15] Error: ENOENT: no such file or directory, stat '/home/travis/build/facebooki

Yeah, we broke the build Friday... in [282f47d](https://github.com/facebookincubator/fbt/commit/282f47d1605c8710e126ba5cc3bad91a50714922) cc @kayhadrin

File: packages/babel-plugin-fbt/FbtUtil.js [hunk 1]
```diff
@@ -116,6 +116,11 @@ function checkOption(option, validOptions, value) {
   return option;
 }
 
+function isBoolAttribute({name}){
```
Comment (jrwats) [inline packages/babel-plugin-fbt/FbtUtil.js]: `canBeShortBoolAttr` is a more appropriate name here.

File: packages/babel-plugin-fbt/FbtUtil.js [hunk 2]
```diff
@@ -116,6 +116,11 @@ function checkOption(option, validOptions, value) {
...
+    return name === 'doNotExtract' || name === 'number'
```
Comment (jrwats) [inline packages/babel-plugin-fbt/FbtUtil.js]: Let's implement this as a static map declared outside the function instead.
Something like the following
```
const keyMirror = require('fbjs/lib/keyMirror');

// ...

const SHORT_BOOL_CANDIDATES = keyMirror({
  doNotExtract: null,
  number: null,
});

function canBeShortBoolAttr(name) { 
  return name in SHORT_BOOL_CANDIDATES;
}
```

**Comment by jrwats** [comment]
Thanks for the tests!  The idea looks good, just asking for some minor tweaks.

**Pull request closed without merging**